### DISCUSSION TITLE
watch method using useFormContext()

### DISCUSSION THREAD

**Discussion by fabiendeborde** [react-hook-form/react-hook-form#1672] [General] [answered]
I am sorry if this is a dumb question, but I expected `watch` to watch an input for any changes it gets but it seems it only works if tis input only is changed. If I update a different input in the same form I lose the watch effect.

```
import React from 'react'
import { useFormContext } from 'react-hook-form'

const PropertyOwnerInfo = () => {
  const { watch } = useFormContext()
  const enabled = watch('property_enabled')
  console.log('enabled', enabled)
  return (
    <div></div>
  )
}

export default PropertyOwnerInfo
```

I have a form in another component with a few inputs. 
If i touch only `property_enabled` watch is working as expected, if I touch another component it stops working.
Am i misunderstanding the way to use watch?

**Comment by bluebill1049** [comment]
> if I touch another component it stops working.

what do you mean?

**Reply by fabiendeborde** [reply]
Sorry if I wasn't clear enough.
I have multiple inputs in my form. In the component I wrote above I watch for the input `property_enabled`. It is a checkbox so if I check/uncheck this input **only**, the `enabled` constant watch correctly for this input.
But if I modify another input, say `property_name`, by adding some text in it, then the constant `enabled` stop watching the `property_enabled` input.

**Reply by bluebill1049** [reply]
That's correct behavior. Watch only watching what you have specified.

**Reply by fabiendeborde** [reply]
I am sorry, my explanation was probably incomplete.
I f I change `property_name` and then change `property_enabled`, the constant `enabled` is not updated.
The watch "effect" stops when any other inputs change.

**Reply by bluebill1049** [reply]
that's not expected, do you have a codesandbox?

**Reply by fabiendeborde** [reply]
I am going to try to create one as soon as I can.

**Reply by fabiendeborde** [reply]
I created a small simplified sample here : https://codesandbox.io/s/muddy-sound-gl47z
and its is working....
I guess the problem comes from the logic I inserted around the inputs and the form...

Thanks a lot for your quick answers! I am going to try to debug this during the weekend, if I find the origin of the problem I'll post here my solution.

**Reply by bluebill1049** [reply]
Please submit an isolated/small codesandbox, it helps me to investigate the problem or potential bug.

**Comment by fabiendeborde** [comment, marked as answer]
I finally identified why the form was stopping watching my input. 
My form was wrapped in a parent container with no access to the form context and was preventing the rerender...

Thanks a lot for your answers, and for asking me to use codesandbox, it really helped me isolating the issue!
By the way, react-hook-form is awesome, keep up this great work!!

**Reply by bluebill1049** [reply]
awesome glad to hear you sort out the problem.
